tests/migration/Core/V6_6/Migration1739355802FixVatHandlingTest.php

```diff
@@ -0,0 +1,73 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Tests\Migration\Core\V6_6;
+
+use Doctrine\DBAL\Connection;
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Test\TestCaseBase\KernelLifecycleManager;
+use Shopware\Core\Migration\V6_6\Migration1739355802FixVatHandling;
+
+/**
+ * @internal
+ */
+#[Package('framework')]
+#[CoversClass(Migration1739355802FixVatHandling::class)]
+class Migration1739355802FixVatHandlingTest extends TestCase
+{
+    private Connection $connection;
+
+    protected function setUp(): void
+    {
+        $this->connection = KernelLifecycleManager::getConnection();
+    }
+
+    public function testUpdateAddsVatPatterns(): void
+    {
+        $this->removeVatPatterns();
```

Comment:
If you use the `DatabaseTransactionBehaviour`, your test is wrapped in an transaction and automatically cleans up its old state afterwards.

Reply:
This doesn't clean up the state. It purposefully removes all vat patterns. I can remove it if you prefer.